ddtrace/internal/telemetry/writer.py

```diff
@@ -0,0 +1,198 @@
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from ...internal import forksafe
+from ...settings import _config as config
+from ..agent import get_connection
+from ..agent import get_trace_url
+from ..compat import get_connection_response
+from ..compat import httplib
+from ..compat import monotonic
+from ..encoding import JSONEncoderV2
+from ..logger import get_logger
+from ..periodic import PeriodicService
+from ..runtime import get_runtime_id
+from ..utils.formats import get_env
+from ..utils.time import StopWatch
+from .data import get_application
+from .data import get_host_info
+
+
+log = get_logger(__name__)
+
+
+def _get_interval_or_default():
+    return float(get_env("instrumentation_telemetry", "interval_seconds", default=60))
+
+
+class TelemetryWriter(PeriodicService):
+    """
+    Periodic service which sends Telemetry request payloads to the agent-proxy
+    """
+
+    ENDPOINT = "telemetry/proxy/api/v2/apmtelemetry"
+
+    def __init__(self, agent_url=""):
+        # type: (str) -> None
+        super(TelemetryWriter, self).__init__(interval=_get_interval_or_default())
+
+        self.enabled = False  # type: bool
+        self.agent_url = agent_url or get_trace_url()
+
+        self._encoder = JSONEncoderV2()
+        self._events_queue = []  # type: List[Dict]
+        self._integrations_queue = []  # type: List[Dict]
+        self._lock = forksafe.Lock()  # type: forksafe.ResetObject
+
+        # _sequence is a counter representing the number of requests sent by the writer
+        self._sequence = 1  # type: int
+
+    def _send_request(self, request):
+        # type: (Dict) -> Optional[httplib.HTTPResponse]
```

Comment:
```suggestion
        # type: (Dict) -> httplib.HTTPResponse
```

I don't see how `None` can be returned here or why we'd want that

Reply:
It will be None if an exception is raised by the http client

Reply:
Yup, I can update the typing. If an response is returned it will never be none